### ISSUE TITLE
Product Page v2.0: page load speed, LCP "needs improvement"

### ISSUE THREAD

**Issue by ifixitmaxb** [iFixit/react-commerce#1015]
[Google Search Console is reporting that all our new pages have high LCP (longer than 2.5s on mobile)](https://search.google.com/search-console/core-web-vitals/summary?resource_id=https%3A%2F%2Fwww.ifixit.com%2Fproducts%2F&device=2)
[Google Search Console was reporting all URLs were "Good" but then as of Nov 4, all pages "Need improvement"](https://search.google.com/search-console/core-web-vitals/summary?resource_id=https%3A%2F%2Fwww.ifixit.com%2Fproducts%2F&device=1), again for high LCP

**Comment by danielbeardsley** [comment]
I've got a way to defer loading youtube videos that I'm working on.

**Comment by danielbeardsley** [comment]
Man, youtube is really hurting us here. I'll see what I can do about that:
![image](https://user-images.githubusercontent.com/26855/202331626-26652e42-7b8f-4030-968c-2557404d9f6e.png)

**Comment by danielbeardsley** [comment]
There's a few small changes that are deployed and a few small ones that are in open pulls. The stats on the product page are still.... not great. 

Note sure where to go from here. 


[The analysis stats on vercel](https://vercel.com/ifixit/react-commerce-prod/analytics?device=mobile&timePeriod=7d&percentile=90&timeInterval=1h&sort=datapoints&search=) are very nice, but the changes we have merged haven't moved that needle at all and the numbers I'm getting locally for mobile lighthouse runs aren't promising (Lots of JS to parse and evaluate).

Potential roads for improvement:

* Enable the old product page (with a certain get param) and we can compare these same stats on the old vs new page
   * Would help with know if we've gone backwards or forwards
* Cache the results of the shopify graphQL query somewhere else cause that's the biggest piece of our server-side time
   * in memory (limited space)
   * on some storage service, Cloudfront backed by S3 (could even re-make the request client-side if the value we got from CF is too old)
      * Imagine exposing this query as an API
   * Have iFixit API make the query and cache the result however we need
* Turn up page caching a bit cause we only have a ~7% hit rate:
![image.png](https://images.zenhubusercontent.com/54ff41794f7a08fbb7f24e2c/6205d798-96ff-40c8-8182-32a8a0c1cc37)
* So much unused JS is downloaded on the product page
  * It's hard to determine which parts are unused. Even when we do, there may not be much recourse as some things are just big and can't be split up (chakra, framer motion, react)
  * Try dynamic imports on components that aren't rendered right away
* Find a way to reduce styling
  * About half of the html downloaded is `<style>` tags which are embedded right next to the components that use them
  * Look into ways to reduce the verbosity or increase the re-use?

**Comment by federicobadini** [comment]
If I can give my two cents here I would say that these are all valid strategies.
However I would just be a little bit patient before moving on in order to give enough time for the core web vitals statistics to settle.

We have done several things in parallel like optimizing images, use the light embed for youtube and other minors to improve performances. 
The mako driver kit page - that was scoring 27 in performance lab test - is now scoring something like 55 to 60. So there has been an improvement, but that is still not visible in core web vitals since they are computed on a 28 days window.

I say this because performance (especially in lab test for mobile) is computed on really low end emulated devices. And this is not a valid representative of your audience. Core web vitals will give on the contrary a real feedback of your user experience

**Comment by danielbeardsley** [comment]
> but that is still not visible in core web vitals since they are computed on a 28 days window.

Yes, I'm aware of this, and they only give us the quantity of pages above a threshold. Vercel's realtime analytics based on real users is much better, giving hourly datapoints on specific pages and metrics at specific percentiles:

![image](https://user-images.githubusercontent.com/26855/205352911-961ff58c-0c8e-4aef-9548-71e9505d77c7.png)

https://vercel.com/ifixit/react-commerce-prod/analytics?device=mobile&timePeriod=7d&percentile=90&timeInterval=1h&sort=datapoints&search=

**Comment by federicobadini** [comment]
Closing since product pages are now all satisfying CWV.

<img width="1511" alt="Screenshot 2023-07-05 at 15 01 33" src="https://github.com/iFixit/react-commerce/assets/7805759/8cd07b36-0b4e-4b2f-9978-527002db50d0">
